openff/toolkit/utils/nagl_wrapper.py

```diff
@@ -61,7 +61,7 @@ def is_available(cls) -> bool:
     def assign_partial_charges(
         self,
         molecule: "Molecule",
-        partial_charge_method: str,
+        partial_charge_method: pathlib.Path | str,
```

Comment:
I'm uncomfortable with this change because the ToolkitRegistry magic relies on ToolkitWrappers having the same API, and this pushes `NAGLToolkitWrapper` out of sync with `OpenEyeToolkitWrapper`, `RDKitToolkitWrapper`, and `BuiltInToolkitWrapper`. I can't name a specific thing that would go wrong by adding this but it invites some ugly edge cases down the road. 

And in practical terms, I think we can contain all the needed changes for our current set of issues to openff-nagl-models for faster turnaround.